kendzi3d-jogl-model-render/src/main/java/kendzi/jogl/model/render/ModelRender.java

```diff
@@ -389,16 +428,17 @@ public void resetMaterials() {
     Material defaultMaterial = new Material();
 
     public void setupDefaultMaterial(GL2 pGL) {
-        setupMaterial2(pGL, defaultMaterial);
+        setupMaterial2(pGL, defaultMaterial, drawTwoSided ? GL.GL_FRONT_AND_BACK : GL.GL_FRONT);
     }
 
-    private void setupMaterial2(GL2 pGl, Material material) {
+    private void setupMaterial2(GL2 pGl, Material material, int sides) {
```

Comment:
Material is send to open gl only if changed. Its done by comparing last used material component with material should be used by model. But now it have new parameter called "sides" it can changed depending on value inside model. If two models share the same materials but have different "sides" value "sides" wont be set to opengl. If sides are needed you need to store "lastSides" value and compare with current

Reply:
Ok, fixed.  It resolves the issue you reported with the brighter fence color in combination with the force two sided menu option.  Can you confirm it?